### ISSUE TITLE
get account current balance

### ISSUE THREAD

**Issue by davlasry** [eshaham/israeli-bank-scrapers#383]
Would be cool to get the current balance for a specific account on top of getting the transactions. Could I work on it?

**Comment by esakal** [comment]
Hi @dadou4243,
Currently the scrapers are a bit limiting because they share the same api for all companies. if we want to extend the data being scraped we need to add support in all scrapers before we can actually merge it to master.

A few months ago I was working on an idea that change the architecture of the scrapers that split the scraping process to smaller units (adapters). See #384 pr. Although you can start playing with it, it is still missing few important parts like overall approval of the concept and documentation. I suggest to wait a few days to get feedback about that PR and it will then be easier to try guide how to proceed with this issue.

- Eran.

**Comment by JavierChames** [comment]
Hi @davlasry 
I'm using this great project and maybe I cant give you a tip if I understand what you mean.
if you want to get the current balance for account...you can get it...
i "expanded" for example the file hapoalims.js under scrapers folder and added to retrieve some more info.
see:
return {
      type: _constants.NORMAL_TXN_TYPE,
      identifier: txn.referenceNumber,
      date: (0, _moment.default)(txn.eventDate, DATE_FORMAT).toISOString(),
      processedDate: (0, _moment.default)(txn.valueDate, DATE_FORMAT).toISOString(),
      originalAmount: isOutbound ? -txn.eventAmount : txn.eventAmount,
      originalCurrency: 'ILS',
      chargedAmount: isOutbound ? -txn.eventAmount : txn.eventAmount,
      description: txn.activityDescription,
      currentbalance: txn.currentBalance,
      status: txn.serialNumber === 0 ? _constants.TRANSACTION_STATUS.PENDING : _constants.TRANSACTION_STATUS.COMPLETED,
      memo
    };
  });
}

i added  :     currentbalance: txn.currentBalance,

so for what you get:
txns: [{
      type: string, // can be either 'normal' or 'installments'
      identifier: int, // only if exists
      date: string, // ISO date string
      processedDate: string, // ISO date string
      originalAmount: double,
      originalCurrency: string,
      chargedAmount: double,
      description: string,
       memo: string, // can be null or empty
      installments: {
        number: int, // the current installment number
        total: int, // the total number of installments
      },
      status: string //can either be 'completed' or 'pending'
    }],
you can add 
let currentbalance = transaction.currentbalance


so I use it in the app.js itself and retrieve the current balance for each transaction...
I hope this helps you...and this is what you wanted..

**Comment by davlasry** [comment]
@JavierChames Thanks Javier.
How can you make sure your changes will be saved after updating the library (npm or yarn update)?

**Comment by JavierChames** [comment]
Hi
you, right, in the past an update, "removed" my updates, so I always save a copy of my local changes.
if the solution is good for you-for me it works-you are welcome to suggest to add it to the original code :-)

**Comment by LiranBri** [comment]
Is there a built-in way to get the current balance?

**Comment by baruchiro** [comment]
I think if we will wait until all the scrapers will support that, it's like a deadlock.

What about adding a `balance` field in the `TransactionsAccount` interface

https://github.com/eshaham/israeli-bank-scrapers/blob/07ecd3de0c4aa051f119aa943493f0cda943158c/src/transactions.ts#L2-L5

With type something like `number | NOT_IMPLEMENTED`?

**Comment by esakal** [comment]
I don't mind having  that field, we can set it to optional property so null means not implemented and number means a balance. 

If you want to create a PR with that field and an implementation in institutes that you have access to feel free. 

@eshaham any objections?

**Comment by baruchiro** [comment]
@esakal Hey, but what about Credit Card? It will never have a balance, am I right?

It can be OK, but just consider it.

**Comment by esakal** [comment]
@baruchiro  note that we added last balance and not running balance for each transation, again I can only help so I do see any reasons not having it there 👍

**Comment by baruchiro** [comment]
I think you need to keep this issue open, because we need to work on the other accounts, so someone else can use this issue to get help.

And note that I don't know exactly in TS, but maybe this type-change is a breaking change and requires `major` version upgrade?

**Comment by esakal** [comment]
@baruchiro  it shouldn't break anything because you added an optional property by setting the ? sign. 

anyhow, I will open a followup ticket with relevant tag to ask people to add their support. 

Regrading credit cards, I think we can use the same property for them as well. the only difference is that you might have more than one balance if you have pending transaction in foreign currencies but for the main currency it should fit, wdyt?

**Comment by esakal** [comment]
lets continue any discussion about it here #547 547
